tests/test_tapify.py

```
@@ -286,7 +499,36 @@ class ConcatDataclass:
             def __eq__(self, other: str) -> bool:
                 return other == concat(self.so, self.few)
 
-        for class_or_function in [concat, Concat, ConcatDataclass]:
+        if _IS_PYDANTIC_V1 is not None:
+
+            @pydantic.dataclasses.dataclass
+            class ConcatDataclassPydantic:
+                so: int
+                few: float
+
+                def __eq__(self, other: str) -> bool:
+                    return other == concat(self.so, self.few)
+
+            class ConcatModel(pydantic.BaseModel):
+                if _IS_PYDANTIC_V1:
+
+                    class Config:
+                        extra = pydantic.Extra.forbid  # by default, pydantic ignores extra arguments
+
+                else:
+                    model_config = pydantic.ConfigDict(extra="forbid")  # by default, pydantic ignores extra arguments
+
+                so: int
+                few: float
+
+                def __eq__(self, other: str) -> bool:
+                    return other == concat(self.so, self.few)
```

Here is a test for the "extra argument should raise an error" functionality